### ISSUE TITLE
Type error when accessing elements of a declared type whose underlying type is a slice

### ISSUE THREAD

**Issue by jcp19** [viperproject/gobra#214]
The following leads to a type error
```go
type IP []byte

func test(ip4 IP) bool {
	return ip4[0] == 127
}
```
Error:
```
IntT(IntegerKind(integer)) index is not a proper index of DeclaredT(type IP []byte,viper.gobra.frontend.info.implementation.TypeInfoImpl@5d970d42)
```

**Comment by jcp19** [comment]
The same occurs for calling `len(s)` where `s` is of a type whose underlying type is a slice

**Comment by jcp19** [comment]
This was fixed by PR #305
